### PR TITLE
[WPC-1329]: Personal data form

### PR DESCRIPTION
**Pull request by antkiewiczk**

### CODE REVIEW

File: libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  ) => void
+} & BuilderContainerProps
+
+export function PersonalDataFormContainer({onUpdate, className}: PersonalDataFormContainerProps) {
+  const {PersonalDataForm} = useWebsiteBuilder()
+  const {user} = useUser()
+
+  const [uploadImage, uploadImageData] = useUploadImageMutation({
+    onCompleted(data) {
+      console.log('data', data)
```
Comment (antkiewiczk) [inline libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx:32]: not entirely sure what to do with these

File: libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  })
+
+  const [updatePassword, updatePasswordData] = useUpdatePasswordMutation({
+    onCompleted(data) {
+      console.log('data', data)
```
Comment (antkiewiczk) [inline libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx:38]: not entirely sure what to do with these

File: libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+  })
+
+  const [updateUser, updateUserData] = useUpdateUserMutation({
+    onCompleted(data) {
+      console.log('data', data)
```
Comment (antkiewiczk) [inline libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx:44]: not entirely sure what to do with these

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+export type ImageUploadProps = {
+  image?: FullImageFragment | null
+  onUpload: (image: any) => void
+}
+
+const Avatar = styled.img`
+  width: 150px;
+  height: 150px;
+  border-radius: 150px;
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:15]:
```suggestion
  border-radius: 100%;
```

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import styled from '@emotion/styled'
+import {FullImageFragment} from '@wepublish/website/api'
+import {useRef} from 'react'
+import {MdDelete, MdEdit} from 'react-icons/md'
+import {IconButton} from '../icon-button/icon-button'
+
+export type ImageUploadProps = {
+  image?: FullImageFragment | null
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:8]: UI shouldn't rely on API models

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+`
+
+const Wrapper = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+`
+
+const ActionButton = styled(IconButton)`
+  margin-left: 10px;
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:25]: Couldn't we use a grid on the parent? Else the sibling/child has to know about the layout.
Also please use the theme.

Note that this is not overrideable now in the website builder, as the IconButton is hardcoded and not taken from the builder (UI can not use the builder due to circular dependencies).

Would this maybe make sense to be in /user/website?

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <Wrapper>
+      <Avatar src={image?.mediumURL || 'https://unsplash.it/300/200'} alt="user-avatar" />
+      <input
+        type="file"
+        accept="image/*"
+        onChange={onUpload}
+        style={{display: 'none'}}
+        id="image-upload-input"
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:39]: use, useId() here please!

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  const fileInputRef = useRef<HTMLInputElement>(null)
+
+  return (
+    <Wrapper>
+      <Avatar src={image?.mediumURL || 'https://unsplash.it/300/200'} alt="user-avatar" />
+      <input
+        type="file"
+        accept="image/*"
+        onChange={onUpload}
+        style={{display: 'none'}}
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:38]: No inline styles please! Also this causes it to fail accessibility tests, as `display: none` removes it from the playback.

File: libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx (hunk 1 above)
Comment (Itrulia) [reply libs/user/website/src/lib/personal-data-form/personal-data-form-container.tsx:32]: can all be removed

File: libs/ui/src/lib/image-upload/image-upload.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import styled from '@emotion/styled'
+import {FullImageFragment} from '@wepublish/website/api'
+import {useRef} from 'react'
+import {MdDelete, MdEdit} from 'react-icons/md'
+import {IconButton} from '../icon-button/icon-button'
+
+export type ImageUploadProps = {
+  image?: FullImageFragment | null
+  onUpload: (image: any) => void
```
Comment (Itrulia) [inline libs/ui/src/lib/image-upload/image-upload.tsx:9]: Could we type this properly?

**Review by Itrulia — APPROVED** [review]
